src/stackAnalysisService.ts

```diff
@@ -68,44 +86,54 @@ export module stackAnalysisServices {
             if (resp.error === undefined && resp.status === 'success') {
               resolve(resp.id);
             } else {
-              errorMsg = `Failed :: ${resp.error}, Status: ${
-                httpResponse.statusCode
-              }`;
+              errorMsg = `Failed :: ${resp.error}, Status: ${httpResponse.statusCode
+                }`;
               reject(errorMsg);
             }
+          } else if (httpResponse.statusCode === 400) {
+            errorMsg = `Manifest file(s) are not proper. Status:  ${httpResponse.statusCode
+              } - ${httpResponse.statusMessage} `;
+            reject(errorMsg);
           } else if (httpResponse.statusCode === 401) {
             clearContextInfo(context);
-            errorMsg = `Failed :: ${httpResponse.statusMessage}, Status: ${
-              httpResponse.statusCode
-            }`;
+            errorMsg = `Failed :: ${httpResponse.statusMessage}, Status: ${httpResponse.statusCode
+              }`;
             reject(errorMsg);
+          } else if (retryCount < MAX_RETRIES) {
```

Comment:
I think you should move the "retry if" below next if. Otherwise, you will retry while getting 403

Reply:
The creators of this code have categorized status code 403 and 429 (in one if statement) as a `server busy` error, message `Service is currently busy to process your request for analysis, please try again in few minutes`.
Im not sure why, but i assume the reasoning comes from the behavior of the server in these cases. 

According to this categorization, shouldn't both status codes in this case be retried?

Reply:
yes